internal/http/services/owncloud/ocs/handlers/apps/sharing/sharees/sharees.go

```diff
@@ -105,9 +105,9 @@ func (h *Handler) userAsMatch(u *userpb.User) *conversions.MatchData {
 		Label: u.DisplayName,
 		Value: &conversions.MatchValueData{
 			ShareType: int(conversions.ShareTypeUser),
-			// TODO(jfd) find more robust userid
-			// username might be ok as it is unique at a given point in time
-			ShareWith: u.Username,
+			// api compatability with oc10: always use the username
+			ShareWith:               u.Username,
+			ShareWithAdditionalInfo: u.Mail,
```

Comment:
oc10 lets the admin decide if the additional info should be `Mail` or `UserID`. For full compatibility you could make the attribute configurable here as well 🤷

Reply:
Maybe later. For now, I just want to be able to distinguish the users. And email is what users know from their email client. Furthermore, for guests the username equals the email. Since we want to unify sharing with _people_ I prefer the mail.

Whin searching recipients, the username is available as the `shareWith`, anyway. The Displayname is in the Label ... which only leaves email ...

Reply:
But with this PR we'll lose the option to hide it. Because ownCloud Web renders the `ShareWithAdditionalInfo` as soon as it exists. Unnecessary info for small instances... 👀

Reply:
I don't think it is unnecessary. I think we should align on emails because they allow using the federated discovery mechanism described by openid connect. phoenix could actually do that instead of using the ocs api ... or ... well ... in addition ...

but that is a product management decision.